docs/linking_to_external_code/integrity_checking.md

```diff
@@ -26,8 +58,20 @@ Collaborator on another machine -- in a freshly cloned project tree:
 ```shell
 # Download the project's dependencies into the machine's cache, integrity
 # checking each resource.
-deno cache -r --lock=lock.json src/deps.ts
+deno cache --reload --lock=lock.json src/deps.ts
 
 # Done! You can proceed safely.
 deno test --allow-read src
 ```
+
+Note that `deno run` will cause dependencies to load and compile if you have not
+cached them first as above. Therefore as a fail safe, you can also use the lock
+file when running your module to validate modified dependencies are not used.
+This will not update the lock file, only validate dependencies in your module
+against those found in the lock file. Use `deno cache` method above as the best
+practice way to update your lock file with missing dependencies.
+
+```shell
+# Run src/deps.ts, validating dependencies against lock.json
+deno run --lock=lock.json src/deps.ts
+```
```

Comment:
Now that I think about it, there is a specific reason you may want to use lock flags with `deno run`.

```suggestion
Pre-populating the cache with `deno cache` as suggested above means that nothing is downloaded at runtime and the integrity flags can be elided. However, this is not the case if dynamic imports are used.  If you are granting net permission to the program, make sure to carry on using the `--lock` flag for runtime commands.

```shell
# Still not safe! We are granting net permission, so pass the --lock flag.
deno run --lock=lock.json --allow-net src/main.ts
deno test --lock=lock.json --allow-net src
```
```

Reply:
So, if I understand correctly, dynamic imports are only downloaded at runtime, and therefore the only way to update them in the lock file is to use the lock write option? E.g.
```shell
deno run --lock=lock.json --lock-write --allow-net my_mod_with_dynamic_imports.ts
```

Reply:
I was focusing on the `--lock` check part of it but yeah, what you said is true.. Honestly I think the type of imports we are talking about should fail when lock checking, rather than requiring to `run` it to write it in the lock file. This can be done with `--cached-only`. I will edit my suggestion above.

Reply:
OK, sounds like there's two use cases then which aren't covered in the PR yet.  The first is dynamic imports you expect in your module and want to be in your lock file (for which you need to run `deno run --lock-file=lock.json --lock-write`) and another use case where you want to protect against new, unexpected, dynamic imports.  I'll update the PR with your suggestion combined with more detail on desired dynamic imports.

Reply:
I forgot to mention: simple dynamic import statements with static strings are treated like static imports and downloaded at compile time. As for the "true" dynamic imports that you need to `run` once to download... it's up to you but my first instinct is to just recommend `--cached-only` to disable them and leave it at that. Imagine if the generated URL was different every time because it was based on the date or something. The lock mechanism just doesn't work well with those.

Reply:
> I forgot to mention: simple dynamic import statements with static strings are treated like static imports and downloaded at compile time.

That should be true only for `deno bundle` command...

Reply:
Perhaps I'm doing this wrong, but just playing around I can't get anything to populate `lock.json` with dynamic imports like this:
```
const child = (await import('https://deno.land/std@v0.50.0/testing/asserts.ts'));
console.log(child);
```
Running with this:
```
deno run --allow-net --allow-read --lock=lock.json --lock-write -r dynamic_import.ts
```
Gives me lock.json of:
```
{}
```
Is this a bug or expected behaviour?  The assert modules are downloaded and compiled, but not locked.

Reply:
Dynamic imports look slightly problematic.  So as a suggested way forward, I've updated the docs with info on `--cached-only` and added a todo for dynamic imports in a further PR.